Xamarin.Forms.Platform.Android/FastRenderers/AccessibilityProvider.cs

```diff
@@ -145,8 +145,7 @@ bool SetHint(string hint = GetFromElement)
 			string value = hint;
 			if (value == GetFromElement)
 			{
-				value = string.Join(". ", (string)Element.GetValue(Accessibility.NameProperty),
-					(string)Element.GetValue(Accessibility.HintProperty));
+				value = string.Join((String.IsNullOrWhiteSpace((string)(Element.GetValue(Accessibility.NameProperty))) || String.IsNullOrWhiteSpace((string)(Element.GetValue(Accessibility.HintProperty)))) ? "" : ". ", (string)Element.GetValue(Accessibility.NameProperty), (string)Element.GetValue(Accessibility.HintProperty));
```

Comment:
This is correct, but it's a very complicated, and quite unreadable, way of writing:
```
var name = (string)Element.GetValue(Accessibility.NameProperty);
var hint = (string)Element.GetValue(Accessibility.HintProperty);
if (string.IsNullOrEmpty(name))
    return hint;
if (string.IsNullOrEmpty(hint))
    return name;
return $"{name}. {hint}"
```